Config/Loader.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Swiftype\SiteSearch\Wordpress\Config;
+
+/**
+ * Configuration loading and management.
+ *
+ * @author Matt Riley <mriley@swiftype.com>, Quin Hoxie <qhoxie@swiftype.com>, Aurelien Foucret <aurelien.foucret@elastic.co>
+ */
+class Loader
+{
+    /**
+     * @var Config
+     */
+    private $config;
+
+    /**
+     * @var Loader
+     */
+    private static $instance;
+
+    /**
+     * Load configuration.
+     *
+     * @param callable $callback Optional function to be called when the configuration is loaded.
+     *
+     * @return \Swiftype\SiteSearch\Wordpress\Config\Loader
+     */
+    public static function loadConfig(callable $callback = null)
+    {
+        if (self::$instance ==null) {
+            self::$instance = (new static());
+        }
+
+        if ($callback !== null) {
+            \add_action('swiftype_config_loaded', $callback);
+        }
+
+        return self::$instance;
+    }
+
+    /**
+     * Constructor.
+     * Use loadConfig method instead.
+     */
+    private function __construct()
+    {
+        $this->config = new Config();
+        \add_action('init', function() {$this->applyUpdates();});
```

Comment:
To be honest I am not very comfortable with this too.

I will try to move that code this way that seems more suitable with WP best practices: https://codex.wordpress.org/Plugin_API/Action_Reference/admin_post_(action)

Reply:
POST data handling is now done into the `Swiftype\SiteSearch\Wordpress\Admin\Action class`:

```
\add_action('admin_action_swiftype_set_api_key', [$this, 'setApiKey']);
\add_action('admin_action_swiftype_create_engine', [$this, 'createEngine']);
\add_action('admin_action_swiftype_clear_config', [$this, 'clearConfig']);
```

The `Swiftype\SiteSearch\Wordpress\Config\Loader` class have been removed from the codebase because these changes make it useless.

Let you share your thoughts on this new implementation.

Commit : d99909c